core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java

```diff
@@ -96,13 +98,20 @@
   private long mStartTimeMs;
 
   /**
+   * The worker ID for this worker. This is set when the block worker is initialized and may be
+   * updated by the block sync thread if the master requests re-registration.
+   */
+  private AtomicReference<Long> mWorkerId;
+
+  /**
    * Constructs a {@link DefaultAlluxioWorker}.
    */
   public DefaultAlluxioWorker() {
+    mWorkerId = new AtomicReference<>();
```

Comment:
I guess this is an artifact of having the registration logic handled in the BlockWorker. If we instead split out a worker client which is responsible for registration, this would be cleaner.